fastestimator/trace/io/csv_logger.py

```diff
@@ -0,0 +1,77 @@
+# Copyright 2019 The FastEstimator Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+from typing import Optional, List
+
+import os
+
+from fastestimator.trace import Trace
+from fastestimator.util.util import is_number, to_list
+from fastestimator.util import Data
+
+
+class CSVLogger(Trace):
+    """Log monitored quantity in CSV file manner
+    Args:
+        filename: Output filename.
+        monitor_names: List of key names to monitor. The names can be
+            {"mode", "epoch", "train_step", or output names that other traces
+            create}. If None, it will record all. Defaults to None.
+        separator: Seperator for numbers. Defaults to ", ".
+        append: If true, it will write csv file in append mode. Otherwise, it
+            will overwrite the existed file. Defaults to False.
+        mode: Restrict the trace to run only on given modes {'train', 'eval',
+            'test'}. None will always execute. Defaults to 'eval'.
+    """
+    def __init__(self,
+                 filename: str,
+                 monitor_names: Optional[List[str]] = None,
+                 separator: str = ", ",
+                 append: bool = False,
+                 mode: str = "eval"):
+        self.keys = monitor_names if monitor_names is None else to_list(monitor_names)
```

Comment:
there should be a way to leverage self.inputs such that self.keys is not necessary